src/Ouroboros-Mini_Protocols.thy

```diff
@@ -149,16 +158,46 @@ print_translation \<open>
   [Syntax_Trans.preserve_binder_abs_tr' \<^const_syntax>\<open>Await\<close> \<^syntax_const>\<open>_Await\<close>]
 \<close>
 
+text \<open>
+  The type \<^type>\<open>program\<close> has only untyped versions of export and import as data constructors,
+  because typed versions would have required support for existential types, which Isabelle/HOL does
+  not have. However, we can introduce typed versions of these constructs by wrapping the untyped
+  ones appropriately. This is analogous to what is done for the send and the receive construct of
```

Comment:
I would say that “construct” is correct here. The intention is that the explicit form of this phrase is “the send construct and the receive construct”, an the phrase in the documentation just leaves out the first occurrence of “construct” to avoid repetition. Leaving out all occurrences but the last is possible in German; I hope that it is also possible in English. What do you think?

Reply:
It would be different if I’d leave out the second “the” and say “the send and receive constructs”. Then the sole remaining “the” would refer to send and receive together, and together they would be multiple constructs, calling for the use of the plural “constructs”. Am I mistaken here?

Reply:
I'm not really sure, it still looks a bit strange to me when reading it. I think we should consult a native English speaker in order to resolve this. :smile: However, I wouldn't like this detail to be a stopper for approving the PR.

Reply:
Do you think it would be correct to say “the send and receive constructs” (no second “the” and plural)?

Reply:
Yes, I think that sounds better. :slightly_smiling_face:

Reply:
I’ve changed this with 3cea883b95d7870a6f17eb71b8ef3fb0ee155337. Can you please check?

Reply:
Looks good now.